apps/gk_species_projection.c

```diff
@@ -139,44 +168,124 @@ gk_species_projection_init(struct gkyl_gyrokinetic_app *app, struct gk_species *
       .mass = s->info.mass,
       .bimaxwellian = bimaxwellian, 
       .divide_jacobgeo = false, // final Jacobian multiplication will be handled in advance
+      .max_iter = max_iter,
+      .eps = iter_eps,
+      .use_last_converged = use_last_converged, 
       .use_gpu = app->use_gpu,
     };
-    proj->proj_max = gkyl_gk_maxwellian_proj_on_basis_inew( &inp_proj );
+    proj->corr_max = gkyl_gk_maxwellian_correct_inew( &inp_corr );
+  }
 
-    proj->correct_all_moms = false; 
-    if (inp.correct_all_moms) {
-      proj->correct_all_moms = true;
+}
+
+static void
+init_maxwellian_gaussian(struct gkyl_gyrokinetic_app *app, struct gk_species *s, 
+  struct gkyl_gyrokinetic_projection inp, struct gk_proj *proj)
+{
+  // Fill the box_size attribute of the projection (used for periodicity).
+  struct func_gaussian_ctx fg_ctx;
+  fg_ctx.f_floor = inp.f_floor;
```

Comment:
maybe change this to 
```
fg_ctx.f_floor = inp.f_floor > 0.0? inp.f_floor : 0.0
```
for safety. I forget how every object is initialized, but I want to avoid the chance that f_floor has an uninitialized (junk) value. I guess it could have a junk, positive, uninitialized value in which case this protection wouldn't work either... dunno